Plugins/AWSLambdaDeployer/Plugin.swift

```diff
@@ -0,0 +1,348 @@
+        guard deployProducts.count == 1,
+              deployProducts[0].targets.count == 1
+        else {
+            throw Errors.deploymentDescriptorProductNotFound("Deploy")
+        }
+        for t in deployProducts[0].targets {
+            print("\(t.name) - \(t.directory)")
+        }
+        guard deployProducts[0].targets.count == 1 else {
+            fatalError("There is more than one Deploy target \(deployProducts[0].targets)")
+        }
+#if arch(arm64)
+        let arch = "arm64-apple-macosx"
+#else
+        let arch = "x86_64-apple-macosx"
+#endif
+        var deployExecutableURL = URL(fileURLWithPath: deployProducts[0].targets[0].directory.string)
+        // while there is /Sources or /Deploy => remove last path component 
+        while deployExecutableURL.path.contains("/Sources") ||
+              deployExecutableURL.path.contains("/Deploy") {
+
+            deployExecutableURL = deployExecutableURL.deletingLastPathComponent()
+        }
+        self.deployExecutable = deployExecutableURL.appendingPathComponent(".build/\(arch)/\(self.buildConfiguration)/Deploy")
+        
+        if self.verboseLogging {
+            print("-------------------------------------------------------------------------")
+            print("configuration")
+            print("-------------------------------------------------------------------------")
+            print(self)
+        }
+    }
+
+    var description: String {
+    """
+    {
+      products: \(self.products.map(\.name))
+      buildConfiguration: \(self.buildConfiguration)
+      deployExecutable: \(self.deployExecutable)
+    }
+    """
+    }
+}
+
+private enum Errors: Error, CustomStringConvertible {
+    case invalidArgument(String)
+    case unsupportedPlatform(String)
+    case unknownProduct(String)
+    case productExecutableNotFound(String)
+    case deploymentDescriptorProductNotFound(String)
+    case processFailed([String], Int32)
+
+    var description: String {
+        switch self {
+        case .invalidArgument(let description):
+            return description
+        case .unsupportedPlatform(let description):
+            return description
+        case .unknownProduct(let description):
+            return description
+        case .productExecutableNotFound(let product):
+            return "product executable not found '\(product)'"
+        case .deploymentDescriptorProductNotFound(let product):
+            return "your project Package.swift has no executable named '\(product)'"
+        case .processFailed(let arguments, let code):
+            return "\(arguments.joined(separator: " ")) failed with code \(code)"
+        }
+    }
+}
+
+// **************************************************************
+// Below this line, the code is copied from the archiver plugin
+// **************************************************************
+
+private enum ProcessLogLevel: Comparable {
```

Comment:
make DRYer with other plugin?

Reply:
I will factor out code common to the two plugin in a shared library

Reply:
I wonder what is the best approach to share code between two Plugins ?  Share a common executable to carry on the actual work ? Symlinks ?

Check  this thread https://forums.swift.org/t/difficulty-sharing-code-between-swift-package-manager-plugins/61690/11

Reply:
yea this is a "hole" in SwiftPM plugins that the team is working to fix afaik, the work around is to put it in a file and symlink it across the targets

Reply:
OK - will do that

Reply:
Done with https://github.com/sebsto/swift-aws-lambda-runtime/commit/8a816c5f34d3aefd4d299b6cde42efa651ad9c02